src/PostgREST/App.hs

```diff
@@ -227,23 +229,28 @@ app dbStructure conf apiRequest =
               let acceptH = (hAllow, if tableInsertable table then "GET,POST,PATCH,DELETE" else "GET") in
               return $ responseLBS status200 [allOrigins, acceptH] ""
 
-        (ActionInvoke, TargetProc qi, Just (PayloadJSON payload)) ->
+        -- These two are ensured by shouldParsePayload in ApiRequest
+        (ActionInvoke{isReadOnly=False}, _, Nothing) -> pure notFound
```

Comment:
Previoulsy we had:
```haskell
   (ActionInvoke, TargetProc qi, Nothing) ->
   -- Not explicitly pattern matched and going to notFound through
   _ -> return notFound
```
I was trying to maintain that mapping though I didn't consider to modify the `relevantPayload` function, I think I have a better solution now, with that these pattern matches can be removed.